### PR TITLE
Add Menu sizes

### PR DESCRIPTION
**Pull request by jgibson02**
## Description

* Adds `extraSmall`, `small`, `large`, and `extraLarge` sizes for Menu components.
* Font size now determined at the Menu component level and adjusted using em units for MenuItem content, Subtitles, and Titles. (Create slight change to previous font sizes)
* Changes Menu to be inline-block and steps the width of the container within the 200 - 280px min-max range.

## Related Issue

In support of #251 

## Motivation and Context

This change provides a greater variety of sizes to fit needs such as context menus.

## How Has This Been Tested?

Visualization via Docz pages.

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: src/Menu/Menu-styled.js [hunk 1]
```diff
@@ -16,15 +16,44 @@ import { unitCalc, fontSize } from '../utils/helpers';
 
 const StyledMenu = styled(StyledSideNav)`
   min-width: 200px;
+  display: inline-block;
```
[the remark this answers is not in the document]
Comment (jgibson02) [reply src/Menu/Menu-styled.js]: This is no longer needed, reverting to `block`.

File: src/Menu/Menu-styled.js [hunk 2]
```diff
@@ -72,6 +101,41 @@ const StyledMenuItem = styled(CalciteA)`
...
+  ${props =>
+    props.extraLarge &&
+    css`
+      line-height: 20px;
+      padding: ${props => unitCalc(props.theme.baseline, 1.5, '/')};
+      font-weight: 300;
+    `};
 `;
 
 const StyledMenuTitle = styled(StyledMenuItem)`
```
[the remark this answers is not in the document]
Comment (jgibson02) [reply src/Menu/Menu-styled.js]: After passing the sizes to the MenuTitle components via the new MenuContext, they automatically got scaled padding. Updated screenshot:
image

[the remark this answers is not in the document]
Comment (codylawson) [reply src/Menu/Menu-styled.js]: I'm not seeing the fix to MenuTitle padding in the commit, are you sure it was included?

**Comment by codylawson** [comment]
Thanks @jgibson02!

**Pull request merged**